app/server/identity/identityMiddleware.ts

```diff
@@ -165,9 +165,9 @@ export const withIdentity: (statusCode?: number) => express.RequestHandler = (
             headers: {
               "X-GU-ID-Client-Access-Token":
                 "Bearer " + idapiConfig.accessToken,
-              [X_GU_ID_FORWARDED_SCOPE]: getScopeFromRequestPathOrEmptyString(
-                req.path
-              ),
+              [X_GU_ID_FORWARDED_SCOPE]:
+                req.header(X_GU_ID_FORWARDED_SCOPE) ||
```

Comment:
At the moment, am I correct in thinking that the value if header if defined will always equal the scope derived from the request path? In which case, what's the motivation of doing this? Optimisation? Future-proofing?

Reply:
so the actual endpoint that this `identityMiddleware.ts` will be intercepting may not itself include `/payment/` e.g. `/api/me/mma` which is general use api endpoint.
So the line in question first tries to use the SCOPE coming from the client-side (i.e. only if this general endpoint is being called from within the payment flow), but falls-back to one inferred from the request URL.

Reply:
I see. And would this header ever not be set?

Reply:
i don't think there are any scenarios currently, but this does ensure the payment specific **api** endpoints for example, always send the right scope, regardless of what the client says.